raft/src/test/java/org/apache/kafka/raft/MockLog.java

```diff
@@ -472,4 +489,110 @@ private EpochStartOffset(int epoch, long startOffset) {
         }
     }
 
+    final class MockRawSnapshotWriter implements RawSnapshotWriter {
+        private final OffsetAndEpoch snapshotId;
+        private ByteBuffer data;
```

Comment:
We could probably use `ByteBufferOutputStream` which already handles expansion.

Reply:
Cool. Yeah, let's use that. So many util goodies.